mx.truffle/mx_truffle.py

```diff
@@ -93,3 +94,58 @@ def _truffle_gate_runner(args, tasks):
     'sldebug' : [sldebug, '[SL args|@VM options]'],
     'testdownstream' : [testdownstream, ''],
 })
+
+"""
+Merges META-INF/truffle/language and META-INF/truffle/instrumentation files.
+This code is tightly coupled with the file format generated by
+LanguageRegistrationProcessor and InstrumentRegistrationProcessor.
+"""
+class TruffleArchiveParticipant:
+    PROPERTY_RE = re.compile(r'(language\d+|instrumentation\d+)(\..+)')
+
+    def _truffle_metainf_file(self, arcname):
+        if arcname == 'META-INF/truffle/language':
+            return 'language'
+        if arcname == 'META-INF/truffle/instrumentation':
+            return 'instrumentation'
+        return None
+
+    def __opened__(self, arc, srcArc, services):
+        self.settings = {}
+        self.arc = arc
+
+    def __add__(self, arcname, contents):
+        metainfFile = self._truffle_metainf_file(arcname)
+        if metainfFile:
+            properties = self.settings.setdefault(metainfFile, {})
+            for line in contents.strip().split(os.linesep):
+                if not line.startswith('#'):
+                    m = TruffleArchiveParticipant.PROPERTY_RE.match(line)
+                    assert m, line
+                    enum = m.group(1)
+                    prop = m.group(2)
+                    properties.setdefault(enum, []).append(prop)
+            return True
+        return False
+
+    def __addsrc__(self, arcname, contents):
+        return False
+
+    def __closing__(self):
+        for metainfFile, properties in self.settings.iteritems():
+            arcname = 'META-INF/truffle/' + metainfFile
+            lines = []
+            counter = 1
+            for enum in sorted(properties.viewkeys()):
+                assert enum.startswith(metainfFile)
+                newEnum = metainfFile + str(counter)
+                counter += 1
+                for prop in properties[enum]:
+                    lines.append(newEnum + prop)
+
+            content = '\n'.join(lines)
```

Comment:
Really? That means the content of the jar file will be different depending on the platform where it was built. Isn't the jar file supposed to be platform independent?

(but probably it doesn't make any difference, the code that reads it should be robust for different line endings anyway...)